modules/nw-kuryr-cleanup.adoc

```diff
@@ -0,0 +1,208 @@
+    done
+}
+----
+. Remove Kuryr finalizers from services:
++
+[source,terminal]
+----
+$ REMFIN services kuryr.openstack.org/service-finalizer
+----
+. Remove all tagged {rh-openstack} Load Balancers from Octavia:
++
+[source,terminal]
+----
+$ for lb in $(openstack loadbalancer list --tags $CLUSTERTAG -f value -c id); do
+    openstack loadbalancer delete --cascade $lb
+done
+----
+. Remove Kuryr finalizers from all KuryrLoadBalancer CRs. This will trigger their deletion:
++
+[source,terminal]
+----
+$ REMFIN kuryrloadbalancers.openstack.org kuryr.openstack.org/kuryrloadbalancer-finalizers
+----
+. Remove Kuryr finalizers from all pods:
++
+[source,terminal]
+----
+$ REMFIN pods kuryr.openstack.org/pod-finalizer
+----
+. Remove Kuryr finalizers from all KuryrPort CRs, as a consequence, KuryrPort CRs will be deleted:
++
+[source,terminal]
+----
+$ REMFIN kuryrports.openstack.org kuryr.openstack.org/kuryrport-finalizer
+----
+. Remove subports created by Kuryr from trunks. Note, it will take a time, since for every subport there is a check if such port is tagged with `$CLUSTERTAG`:
++
+[source,terminal]
+----
+$ read -ra trunks <<< $(python -c "import openstack; n = openstack.connect().network; print(' '.join([x.id for x in n.trunks(any_tags='$CLUSTERTAG')]))")
+$ i=0
+$ for trunk in "${trunks[@]}"; do
+    i=$((i+1))
+    echo "Processing trunk $trunk, ${i}/${#trunks[@]}."
+    subports=()
+    for subport in $(python -c "import openstack; n = openstack.connect().network; print(' '.join([x['port_id'] for x in n.get_trunk('$trunk').sub_ports if '$CLUSTERTAG' in n.get_port(x['port_id']).tags]))"); do 
+        subports+=("$subport");
+    done
+    args=()
+    for sub in "${subports[@]}" ; do
+        args+=("--subport $sub")
+    done
+    if [ ${#args[@]} -gt 0 ]; then
+        openstack network trunk unset ${args[*]} $trunk
+    fi
+done
+----
+. Get all networks and subnets from KuryrNetwork CRs and remove ports, router interfaces and network itself:
++
+[source,terminal]
+----
+$ mapfile -t kuryrnetworks < <(oc get kuryrnetwork -A --template='{{range $i,$p := .items}}{{ $p.status.netId }}|{{ $p.status.subnetId }}{{"\n"}}{{end}}')
+$ i=0
+$ for kn in "${kuryrnetworks[@]}"; do
+    i=$((i+1))
+    netID=${kn%%|*}
+    subnetID=${kn##*|}
+    echo "Processing network $netID, ${i}/${#kuryrnetworks[@]}"
+    # Remove all ports from the network.
+    for port in $(python -c "import openstack; n = openstack.connect().network; print(' '.join([x.id for x in n.ports(network_id='$netID') if x.device_owner != 'network:router_interface']))"); do
+        ( openstack port delete $port ) &
+
+        # Only allow 20 jobs in parallel.
+        if [[ $(jobs -r -p | wc -l) -ge 20 ]]; then
+            wait -n
+        fi
+    done
+    wait
+
+    # Remove the subnet from the router.
+    openstack router remove subnet $ROUTERID $subnetID
+
+    # Remove the network.
+    openstack network delete $netID
+done
+----
+
+. Remove Kuryr service network:
```

Comment:
Perhaps this could at the step that deletes services as this is related to services